contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -161,6 +185,44 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
         super.revokeRole(role, account);
     }
 
+    /**
+     * @dev Returns the amount of seconds to wait after the `newDefaultAdminDelay` will
+     * become the new {defaultAdminDelay}.
+     *
+     * The value returned guarantees that if the delay is reduced, it will go into effect
+     * after a wait that honors the previously set delay.
+     *
+     * See {defaultAdminDelayIncreaseWait}.
+     */
+    function _delayChangeWait(uint48 newDefaultAdminDelay) internal view returns (uint48) {
+        uint48 currentDelay = defaultAdminDelay();
+
+        // Schedules defaultAdminDelayIncreaseWait() if the delay is increased, this is done so the user has time enough to fix an accidentally high new delay set.
+        // If the delay is reduced, wait the difference between current and new delay to guarantee the delay change schedule + a default admin change
+        // is effectively the current delay. For example, if delay is reduced from 10 days to 3 days, it's needed to wait 7 days
+        // before starting the new 3 days delayed transfer summing up to 10 days, which is the current delay.
+        return
+            newDefaultAdminDelay > currentDelay ? defaultAdminDelayIncreaseWait() : currentDelay - newDefaultAdminDelay;
+    }
+
+    /**
+     * @dev Setter of the tuple for pending admin and its schedule.
+     *
+     * May emit a DefaultAdminTransferCanceled event.
+     */
+    function _setPendingDefaultAdmin(address newAdmin, uint48 newSchedule) internal virtual {
```

Comment:
Should we consider a `_setPendingDelay`? It'll be used only once but it makes sense to me for keeping a consistent `internal` API, I just don't know if that's important.

Reply:
Changed it, looks cleaner.

Reply:
![image](https://user-images.githubusercontent.com/33379285/226711415-13ebd376-34c8-46f6-9733-a07673d6146f.png)